app/src/main/java/org/p2p/wallet/common/EncryptedSharedPreferences.kt

```diff
@@ -0,0 +1,203 @@
+package org.p2p.wallet.common
+
+import androidx.core.content.edit
+import android.content.SharedPreferences
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import timber.log.Timber
+import kotlin.properties.ReadWriteProperty
+import kotlin.reflect.KClass
+import kotlin.reflect.KProperty
+import org.p2p.core.crypto.Hex
+import org.p2p.wallet.common.crypto.keystore.DecodeCipher
+import org.p2p.wallet.common.crypto.keystore.EncodeCipher
+import org.p2p.wallet.common.crypto.keystore.KeyStoreWrapper
+
+class EncryptedSharedPreferences(
```

Comment:
instead of randomly using AccountStorage / SecureStorage / smth else we have EncryptedSharedPreferences that has all the logic that encodes / decodes data that comes to shared prefs
no logic inside regarding prefs / namings / cipher is touched.

Reply:
the architecture scheme is on the way